diff -ur a/arch/riscv/boot/dts/allwinner/sun20i-d1.dtsi b/arch/riscv/boot/dts/allwinner/sun20i-d1.dtsi
--- a/arch/riscv/boot/dts/allwinner/sun20i-d1.dtsi	2022-11-02 23:49:55.000000000 -0500
+++ b/arch/riscv/boot/dts/allwinner/sun20i-d1.dtsi	2023-01-02 00:11:31.688713399 -0600
@@ -265,6 +266,12 @@
 				pins = "PG8", "PG9";
 				function = "uart1";
 			};
+
+			/omit-if-no-ref/
+			w1_pin: w1_pin@0 {
+				pins = "PB7";
+				function = "gpio_in";
+			};
 		};
 
 		pwm: pwm@2000c00 {
diff -ur a/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts b/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts
--- a/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts	2022-11-02 23:49:55.000000000 -0500
+++ b/arch/riscv/boot/dts/allwinner/sun20i-d1-mangopi-mq-pro.dts	2023-01-02 14:42:33.775607066 -0600
@@ -73,6 +73,14 @@
 		compatible = "mmc-pwrseq-simple";
 		reset-gpios = <&pio 6 17 GPIO_ACTIVE_LOW>; /* PG17 */
 	};
+
+	onewire@0 {
+		compatible = "w1-gpio";
+		gpios = <&pio 1 7 GPIO_ACTIVE_HIGH>; /* PB7 */
+		pinctrl-0 = <&w1_pin>;
+		pinctrl-names = "default";
+   status="okay";
+	};
 };
 
 &cpu0 {
